backend/urls.py

```diff
@@ -9,8 +9,8 @@
     # API v2
     url(r'^v2/experiments/$',
         api_views.experiments, name='v2-experiments'),
-    url(r'^v2/experiments/(?P<exp_id>\d+)/$',
-        api_views.experiment_by_id, name='v2-experiment-by-id'),
+    url(r'^v2/experiments/(?P<exp_slug>[-\w]+)/$',
```

Comment:
You can also use something like this, from the Django examples, which may be better b/c it will follow what `SlugField` accepts:

        path('articles/<int:year>/<int:month>/<slug:slug>/', views.article_detail),

Reply:
Nice! The `path` approach seems much cleaner. I will update the rest also.